test/test_um2netcdf.py

```diff
@@ -206,124 +212,130 @@ def fake_out_path():
-        m_iris_load.return_value = [ta_plev_cube, geo_potential_cube]
-        m_saver().__enter__ = mock.Mock(name="mock_sman")
+        m_time_coord.side_effect = mock_fix_time_no_time_dim
 
-        # all cubes should be dropped
-        assert um2nc.process(fake_in_path, fake_out_path, std_args) == []
+    # all cubes should be dropped
+    assert list(um2nc.process_cubes(cubes, mule_vars, std_args)) == []
 
 
 def test_process_mask_with_heaviside(ta_plev_cube, precipitation_flux_cube,
                                      heaviside_uv_cube, heaviside_t_cube,
                                      geo_potential_cube, mule_vars,
                                      std_args, fake_in_path, fake_out_path):
     """Run process() with pressure level masking cubes present."""
+
+    # air temp requires heaviside_uv & geo_potential_cube requires heaviside_t
+    # masking, include both to enable code execution for both masks
+    cubes = [ta_plev_cube, precipitation_flux_cube, geo_potential_cube,
+             heaviside_uv_cube, heaviside_t_cube]
+
+    # mock fix_time_coord to avoid adding difficult to replicate
+    # iris methods in DummyCube.
     with (
-        mock.patch("mule.load_umfile"),
-        mock.patch("umpost.um2netcdf.process_mule_vars") as m_mule_vars,
-        mock.patch("iris.load") as m_iris_load,
-        mock.patch("iris.fileformats.netcdf.Saver") as m_saver,  # prevent I/O
         mock.patch("umpost.um2netcdf.apply_mask"),  # TODO: eventually call real version
-        mock.patch("umpost.um2netcdf.cubewrite"),
+        mock.patch("umpost.um2netcdf.fix_fill_value"),
+        mock.patch("umpost.um2netcdf.fix_time_coord") as m_time_coord,
     ):
-        m_mule_vars.return_value = mule_vars
-
-        # air temp requires heaviside_uv & geo_potential_cube requires heaviside_t
-        # masking, include both to enable code execution for both masks
-        cubes = [ta_plev_cube, precipitation_flux_cube, geo_potential_cube,
-                 heaviside_uv_cube, heaviside_t_cube]
-
-        m_iris_load.return_value = cubes
-        m_saver().__enter__ = mock.Mock(name="mock_sman")
+        m_time_coord.side_effect = mock_fix_time_no_time_dim
 
         # all cubes should be processed & not dropped
-        processed = um2nc.process(fake_in_path, fake_out_path, std_args)
-        assert len(processed) == len(cubes)
+        processed = list(um2nc.process_cubes(cubes, mule_vars, std_args))
+
+    assert len(processed) == len(cubes)
 
-        for pc in processed:
-            assert pc in cubes
+    cube_names = {cube.name() for cube in cubes}
+    processed_names = {cube.name() for cube, _, _ in processed}
+    assert processed_names == cube_names
 
 
 def test_process_no_masking_keep_all_cubes(ta_plev_cube, precipitation_flux_cube,
                                            geo_potential_cube, mule_vars, std_args,
                                            fake_in_path, fake_out_path):
```

Comment:
```suggestion
def test_process_no_masking_keep_all_cubes(ta_plev_cube, precipitation_flux_cube,
                                           geo_potential_cube, mule_vars, std_args):
```

Remove unused fixtures